pkg/reconciler/identity/reconciler.go

```diff
@@ -92,6 +92,9 @@ func (i *Identity) ReconcileWorkloadIdentity(ctx context.Context, projectID stri
 // if this k8s service account only has one ownerReference.
 func (i *Identity) DeleteWorkloadIdentity(ctx context.Context, projectID string, identifiable duck.Identifiable) error {
 	status := identifiable.IdentityStatus()
+	if status.ServiceAccountName == "" {
```

Comment:
add a comment... something like
// if the ServiceAccountName wasn't set in the status, it means that no k8s account was created. We simply return